lib/url.js

```diff
@@ -407,33 +407,22 @@ Url.prototype.parse = function(url, parseQueryString, slashesDenoteHost) {
 };
 
 function validateHostname(self, rest, hostname) {
-  for (var i = 0, lastPos; i <= hostname.length; ++i) {
-    var code;
-    if (i < hostname.length)
-      code = hostname.charCodeAt(i);
-    if (code === 46/*.*/ || i === hostname.length) {
-      if (i - lastPos > 0) {
-        if (i - lastPos > 63) {
-          self.hostname = hostname.slice(0, lastPos + 63);
-          return '/' + hostname.slice(lastPos + 63) + rest;
-        }
-      }
-      lastPos = i + 1;
-      continue;
-    } else if ((code >= 48/*0*/ && code <= 57/*9*/) ||
-               (code >= 97/*a*/ && code <= 122/*z*/) ||
-               code === 45/*-*/ ||
-               (code >= 65/*A*/ && code <= 90/*Z*/) ||
-               code === 43/*+*/ ||
-               code === 95/*_*/ ||
-               code > 127) {
-      continue;
-    }
+  for (var i = 0; i <= hostname.length; ++i) {
+    const code = hostname.charCodeAt(i);
```

Comment:
The conditional that previously preceded this needs to be kept, otherwise there could still be a deopt (at least in node v6.x) when trying to access an index that is out of bounds (when `i === hostname.length`).